### PR TITLE
beautified code xsns 33 qmc5883l

### PR DESCRIPTION
**Pull request by fb-pilot**
## Description:
beautified code and save about 90 bytes
**Related issue (if applicable):

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.6
  - [ ] The code change is tested and works with Tasmota core ESP32 V.3.0.0
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino [hunk 1]
```diff
@@ -24,19 +24,20 @@
...
- *
- * Calibration :
- * valid relations :      1 T = 10000 G ... 1G = 0.1mT = 100µT
- * Dynamic Output Field Range ±2 or ±8 Gauss ==> ±200µT or ±800µT
- * Sensitivity [1]
- *   Field Range = ±2G 12000 LSB/G  --> factor = 120
- *   Field Range = ±8G 3000 LSB/G  --> factor = 30
+ * #define QMC5883L_OVERSAMPLE  0     // 0 .. 3 => over Sample Ratio : 512, 256, 128, 64
+ * #define QMC5883L_GAUSS       1     // 0 .. 1 => full Scale : 2GAUSS, 8GAUSS
+ * #define QMC5883L_FILTER      0     // 0 .. 3 => Output Data Rate : 10HZ, 50HZ, 109HZ, 200HZ
```
Comment (eku) [inline tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino:29]: Functional change?

Comment (fb-pilot) [reply tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino:29]: no ... unchanged

Comment (eku) [reply tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino:29]: 0 => 2 ?

Comment (fb-pilot) [reply tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino:29]: an additional compile option : #define MC5883L_DECIMAL to modify displaying dcimals

Comment (arendst) [reply tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino:29]: As eku said: 2 => 0 ? and 1 => 0 ?

File: tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
   xsns_33_qmc5883l.ino - QMC5883L 3-Axis Digital Compass sensor support for Tasmota
 
-  Copyright (C) 2022  Helge Scheunemann and Friedbert Bader
+  Copyright (C) 2022  Helge Scheunemann + fb-pilot 2024-2-18,2024-4-29
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino:4]: Stop messing with the copyright message using dates - No-GO!!

File: tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino (hunk 1 above)
Comment (fb-pilot) [reply tasmota/tasmota_xsns_sensor/xsns_33_qmc5883l.ino:29]: the driver reads the registers once a second - so providing them 10 times a second is more than sufficient (2=>0). That reduces the power consumption of the chip and allows longer filtering (1=<0)

**Comment by fb-pilot** [comment]
you are right - I had to adapt defaults for bf-xsns_33_qmc5883l

File: tasmota/my_user_config.h [hunk 3]
```diff
@@ -728,9 +728,9 @@ An online tool to calculate TLS fingerprints is available here at:\n\
 //  #define USE_QMC5883L                           // [I2CDriver71] Enable QMC5883L magnetic induction sensor (I2C address 0x0D) (+0k8 code)
 //  #define USE_HMC5883L                           // [I2CDriver73] Enable HMC5883L magnetic induction sensor (I2C address 0x1E) (+1k3 code)
 //    #define QMC5883L_TEMP_SHIFT       23         // sensor temperature are not calibrated (only relativ measurement) and need an absolute ground value in °C (see datasheet)
-//    #define QMC5883L_OVERSAMPLE       1          // 0 .. 3 => 512, 256(default), 128, 64
-//    #define QMC5883L_GAUSS            0          // 0,1(default) => 2GAUSS, 8GAUSS(default)
-//    #define QMC5883L_FILTER           2          // 0 .. 3 => 10HZ, 50HZ, 109HZ(default), 200HZ
+//    #define QMC5883L_OVERSAMPLE       0          // 0 .. 3 => 512(default), 256, 128, 64
+//    #define QMC5883L_GAUSS            1          // 0,1(default) => 2GAUSS, 8GAUSS(default)
+//    #define QMC5883L_FILTER           0          // 0 .. 3 => 10HZ(default), 50HZ, 109HZ, 200HZ
```
Comment (eku) [inline tasmota/my_user_config.h:733]:
```suggestion
//    #define QMC5883L_FILTER           0          // 0 .. 3 => 10HZ(default), 50HZ, 100HZ, 200HZ
```

**Review by eku — CHANGES_REQUESTED** [review]

**Pull request merged**